src-self-hosted/codegen/arm.zig

```diff
@@ -389,12 +429,36 @@ pub const Instruction = union(enum) {
                 .rn = rn.id(),
                 .rd = rd.id(),
                 .offset = offset.toU12(),
-                .l = load_store,
-                .w = writeback,
-                .b = byte_word,
-                .u = up_down,
-                .p = pre_post,
-                .i = if (offset == .Immediate) 0 else 1,
+                .load_store = load_store,
+                .write_back = if (write_back) 1 else 0,
```

Comment:
```suggestion
                .write_back = @boolToInt(write_back),
```

Reply:
I was thinking of that as well. Sometimes, the bits represent the intuition, sometimes they are the opposite of the intuition:

https://github.com/ziglang/zig/blob/99a9590a165beb2f73e2644fa9bf43fce984fbcd/src-self-hosted/codegen/arm.zig#L405

https://github.com/ziglang/zig/blob/99a9590a165beb2f73e2644fa9bf43fce984fbcd/src-self-hosted/codegen/arm.zig#L437

In order to be consistent, should I use `@boolToInt` everywhere and invert the argument where necessary?

Reply:
I don't think `if (offset == .Immediate) 0 else 1` is any clearer than `@boolToInt(offset != .Immediate)` or `@boolToInt(offset == .Register)` but if you want to be more explicit then using ifs is fine too.

Reply:
Hmm yeah, `@boolToInt` is probably the better solution here.